.circleci/config.yml

```diff
@@ -0,0 +1,246 @@
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Install local pip packages
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            paver test_system -s cms --with-flaky --cov-args="-p" --with-xunitmp
+
+# ########## LIB UNIT TESTS ##########
+
+  lib_unit_tests:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    environment:
+      - NO_PREREQ_INSTALL: "true"
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Install local pip packages
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            paver test_lib --with-flaky --cov-args="-p" --with-xunitmp
+
+# ########## JAVASCRIPT TESTS ##########
+
+  javascript_tests:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    environment:
+      - NO_PREREQ_INSTALL: "true"
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
```

Comment:
@symbolist nit: could we change it a bit so that it's more descriptive? or add a comment above it? (something like: Install firefox and node for js tests"

Reply:
The packages in `requirements/system/ubuntu/apt-packages.txt` are needed for a number of python packages as well. I can change it to "Install deb packages" to make it consistent with the descriptions of the other install steps. Will that do?